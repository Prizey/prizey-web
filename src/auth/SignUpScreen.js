import React from 'react'
import { Redirect } from '@reach/router'
import { SignUp } from 'croods-auth'
import { SignUp as SignUpForm } from 'seasoned-auth-forms-web'

import Layout from 'components/Layout/Layout'

export const Form = ({ create, creating, error }) => (
  <Layout>
    <SignUpForm onSubmit={create} submitting={creating} submitError={error} />
  </Layout>
)

export default props => {
  if (props.currentUser) return <Redirect to="/" noThrow />

  return (
    <SignUp
      {...props}
      render={renderProps => <Form {...renderProps} />}
      renderCreated={() => <Redirect to="/" noThrow />}
    />
  )
}
