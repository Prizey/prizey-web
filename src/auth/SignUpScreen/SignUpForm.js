import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import { Typography } from '@material-ui/core'
import { SignUp } from 'seasoned-auth-forms-web'

import SubmitButton from 'design/SubmitButton/SubmitButton'
import SecondaryLink from 'design/SecondaryLink'

const styles = theme => ({
  links: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: theme.spacing.sm,
  },
  message: {
    marginTop: theme.spacing.xs,
  },
})

export const FormBottom = withStyles(styles)(({ classes, ...props }) => (
  <React.Fragment>
    <SubmitButton {...props} label="REGISTER" />
    <Typography align="center" className={classes.message}>
      By proceeding, you agree you&apos;re 18+ and agree to the terms of
      service.
    </Typography>

    <div className={classes.links}>
      <SecondaryLink to="/sign-in" label="Already have an account?" />
    </div>
  </React.Fragment>
))

export default props => (
  <SignUp
    {...props}
    renderButton={buttonProps => <FormBottom {...buttonProps} />}
  />
)
