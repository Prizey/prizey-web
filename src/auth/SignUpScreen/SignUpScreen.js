import React from 'react'
import { Redirect } from '@reach/router'
import { SignUp } from 'croods-auth'

import SignUpForm from './SignUpForm'
import AuthForm from '../AuthForm'

export default props => {
  if (props.currentUser) return <Redirect to="/" noThrow />

  return (
    <SignUp
      {...props}
      render={AuthForm(SignUpForm, 'Create your account!')}
      renderCreated={() => <Redirect to="/" noThrow />}
    />
  )
}
