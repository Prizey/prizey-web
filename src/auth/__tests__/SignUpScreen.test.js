import React from 'react'
import renderer from 'react-test-renderer'

import SignUpScreen from '../SignUpScreen'

jest.mock('croods-auth', () => ({
  SignUp: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ user: { id: 1 } })}</div>
    </div>
  ),
}))

jest.mock('seasoned-auth-forms-web', () => ({
  SignUp: props => <div {...props}>SignUp</div>,
}))

jest.mock('@reach/router', () => ({
  Link: props => <div {...props}>Link - {props.children}</div>,
  Redirect: props => <div {...props}>Redirect - {props.children}</div>,
}))

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ id: 1 })}</div>
    </div>
  ),
}))

it('renders correctly', () => {
  const tree = renderer.create(<SignUpScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirect correctly', () => {
  const params = {
    currentUser: true,
  }
  const tree = renderer.create(<SignUpScreen {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})
