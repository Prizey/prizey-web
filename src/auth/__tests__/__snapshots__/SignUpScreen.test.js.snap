// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirect correctly 1`] = `
<div
  noThrow={true}
  to="/"
>
  Redirect - 
</div>
`;

exports[`renders correctly 1`] = `
<div
  render={[Function]}
  renderCreated={[Function]}
>
  New - 
  <div>
    render - 
    <div
      className="Component-root-83"
    >
      <div
        className="Component-header-81"
      />
      <div
        className="Component-container-78"
      >
        <div
          style={
            Object {
              "textDecoration": "none",
            }
          }
          to="/"
        >
          Link - 
          <img
            alt="Prizey"
            className="Component-logoImage-82"
            src="/logo.png"
          />
        </div>
        <div
          className="Component-caption-76"
        />
        <div
          className="Component-children-77"
        >
          <h5
            className="MuiTypography-root-32 MuiTypography-h5-48 MuiTypography-alignCenter-55 Component-pageTitle-75"
          >
            Create your account!
          </h5>
          <div
            renderButton={[Function]}
          >
            SignUp - 
            <button
              className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-contained-14 MuiButton-containedPrimary-15 MuiButton-raised-17 MuiButton-raisedPrimary-18 MuiButton-fullWidth-28"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              renderButton={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-4"
              >
                REGISTER
              </span>
              <span
                className="MuiTouchRipple-root-68"
              />
            </button>
            <p
              className="MuiTypography-root-32 MuiTypography-body1-41 MuiTypography-alignCenter-55 Component-message-2"
            >
              By proceeding, you agree you're 18+.
            </p>
            <div
              className="Component-links-1"
            >
              <div
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                to="/sign-in"
              >
                Link - 
                <span
                  className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorPrimary-62"
                >
                  Already have an account?
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="Component-footer-79"
        >
          <a
            className="Component-footerLink-80"
            href="https://app.termly.io/document/terms-and-conditions/4de5f3bb-a333-47bb-9daf-f82f7d4e1cfb"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-80"
            href="https://app.termly.io/document/privacy-policy/8192f107-a2dc-4548-bf83-97907c6d83c3"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-80"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
            >
              FAIRNESS
            </span>
          </a>
          <a
            className="Component-footerLink-80"
            href="/all-prizes"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorTextSecondary-65 MuiTypography-alignCenter-55"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    renderCreated - 
    <div
      noThrow={true}
      to="/"
    >
      Redirect - 
    </div>
  </div>
</div>
`;

exports[`renders the form correctly 1`] = `
Array [
  <button
    className="MuiButtonBase-root-29 MuiButton-root-3 MuiButton-contained-14 MuiButton-containedPrimary-15 MuiButton-raised-17 MuiButton-raisedPrimary-18 MuiButton-fullWidth-28"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-4"
    >
      REGISTER
    </span>
    <span
      className="MuiTouchRipple-root-68"
    />
  </button>,
  <p
    className="MuiTypography-root-32 MuiTypography-body1-41 MuiTypography-alignCenter-55 Component-message-2"
  >
    By proceeding, you agree you're 18+.
  </p>,
  <div
    className="Component-links-1"
  >
    <div
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/sign-in"
    >
      Link - 
      <span
        className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-colorPrimary-62"
      >
        Already have an account?
      </span>
    </div>
  </div>,
]
`;
